CMakeLists.txt

```diff
@@ -15,7 +15,7 @@ set( CMAKE_INCLUDE_CURRENT_DIR ON )
 include_directories( program/src/ )
 
 # gcc compiler/linker flags
-add_compile_options( -ggdb -Wall -Wextra -Werror -m64 )
+add_compile_options( -ggdb -Wall -Wextra -Werror -Wno-unused-function -m64 )
```

Comment:
What's this for? If actually needed, it would be preferable to mark these functions with `[[maybe_unused]]`, etc., rather than disabling the warning globally.